source/vir/src/split_expression.rs

```diff
@@ -387,16 +388,59 @@ pub(crate) fn split_expr(ctx: &Ctx, state: &State, exp: &TracedExp, negated: boo
             return Arc::new(split_traced);
         }
         ExpX::Bind(bnd, e1) => {
-            let new_bnd = match &bnd.x {
-                BndX::Let(..) if !negated => bnd.clone(),
-                // TODO(channy1413): split quantifiers
+            match &bnd.x {
+                BndX::Let(..) if !negated => (), // REVIEW: Can we support `Let` in negated position?
+                BndX::Quant(Quant { quant: air::ast::Quant::Forall, boxed_params: _ }, _, _)
+                    if !negated =>
+                {
+                    ()
+                }
+                BndX::Quant(Quant { quant: air::ast::Quant::Exists, boxed_params: _ }, _, _)
+                    if negated =>
+                {
+                    ()
+                }
                 _ => return mk_atom(exp.clone(), negated),
             };
+
+            let new_bnd = |e1: &Exp| -> Bnd {
+                match &bnd.x {
+                    BndX::Let(..) if !negated => bnd.clone(),
+                    // Note: trigger selection happens after the AST->SST translation(see crate::ast_to_sst::finalized_exp)
+                    BndX::Quant(
+                        Quant { quant: air::ast::Quant::Forall, boxed_params },
+                        bndrs,
+                        _trigs,
+                    ) if !negated => {
+                        let bndx = BndX::Quant(
+                            Quant { quant: air::ast::Quant::Forall, boxed_params: *boxed_params },
+                            shrink_binders(e1, bndrs),
+                            Arc::new(vec![]), // remove triggers that are already selected
```

Comment:
This would be better just keeping the original binders and the original triggers.  (Same for Exists.)